File: docs/archive/ScotlandsMountains/Domain/Mountain.cs
namespace ScotlandsMountains.Domain;

public class Mountain : Entity
{
    public List<string> Aliases { get; set; }

    public Location Location { get; set; }

    public int DobihId { get; set; }

    public string GridRef { get; set; }

    public Height Height { get; set; }

    public Prominence Prominence { get; set; }

    public string Features { get; set; }

    public string Observations { get; set; }

    public MountainSummary? Parent { get; set; }

    public EntitySummary Section { get; set; }

    public List<EntitySummary> Counties { get; set; }

    public List<ClassificationSummary> Classifications { get; set; }

    public List<EntitySummary> Maps { get; set; }
}

File: docs/archive/ScotlandsMountains/UI/src/app/models/mountain.ts
import { ClassificationSummary } from "./classification"
import { CountySummary } from "./county";
import { MapSummary } from "./map";
import { SectionSummary } from "./section";

export interface MountainSummary {
    id: string;
    name: string;
    height: number;
    longitude: number;
    latitude: number;
}

export interface Mountain {
    id: string;
    name: string;
    aliases: string[];
    gridRef: string;
    height: number;
    longitude: number;
    latitude: number;
    prominence: number;
    prominenceFrom: string;
    prominenceFromHeight: number;
    features: string;
    observations: string;
    parent: MountainSummary;
    section: SectionSummary;
    counties: CountySummary[];
    classifications: ClassificationSummary[];
    maps: MapSummary[];
}
